var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -246,6 +248,15 @@ class Mfem(Package):
     depends_on('umpire@2.0.0:', when='+umpire')
     depends_on('umpire+cuda', when='+umpire+cuda')
 
+    depends_on('amgx', when='+amgx')
+    # AmgX doesn't need CUDA but MFEM build system requires amgx+cuda
```

Comment:
pretty sure this is incorrect :)

Reply:
It's been a little while since I wrote this but I think it was based on the cusparse/cublas in MFEM's `AMGX_LIB`.  It looks like the `~cuda` variant of `amgx` is [broken/useless](https://github.com/spack/spack/blob/e75b76f433016a720027b41c0d36cdb9ef24a827/var/spack/repos/builtin/packages/amgx/package.py#L47-L48) anyways, so I've added a conflict for `+amgx~cuda` per Veselin's suggestion.

Reply:
I don't think CUDA is optional for AmgX. Their README.md, https://github.com/NVIDIA/AMGX/blob/main/README.md, says: "In order to build project you would need CMake and CUDA Toolkit." So having `cuda` variant in the `amgx` package seems wrong.

Reply:
I agree with that assessment - the `WITH_CUDA` flag that the `amgx` package attempts to set does not exist in their build system from what I can tell.